syntax = "proto3";

package securecloud.auth;

// Service d'authentification
service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc RefreshToken (RefreshTokenRequest) returns (RefreshTokenResponse);
  rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenResponse);
  rpc RevokeTokens (RevokeTokensRequest) returns (RevokeTokensResponse);
}

// Messages
message LoginRequest {
  string username = 1;
  string password = 2;
  string mfa_code = 3;
}

message LoginResponse {
  string access_token = 1;
  string refresh_token = 2;
  int64 expires_in = 3;
}

message RefreshTokenRequest {
  string refresh_token = 1;
}

message RefreshTokenResponse {
  string access_token = 1;
  int64 expires_in = 2;
}

message ValidateTokenRequest {
  string access_token = 1;
}

message ValidateTokenResponse {
  bool valid = 1;
  string user_id = 2;
  repeated string permissions = 3;
}

message RevokeTokensRequest {
  string user_id = 1;
}

message RevokeTokensResponse {
  bool success = 1;
}
